src/core/store.ts

```diff
@@ -177,6 +177,7 @@ export const createStore = (
       return
     }
     atomState.c?.() // cancel read promise
+    delete atomState.e // read error
```

Comment:
This was being done when writing to a sync atom, but not an async atom.

Reply:
Wow! Nice catch!
Feels like you have fairly deeper understanding of core.